src/views/datasets/buoy-telemetry/summary.vue

```diff
@@ -0,0 +1,327 @@
+        <div class="video-wrapper mb-3">
+          <iframe
+            src="https://www.youtube.com/embed/uPXvtlrC7Rs"
+            title="YouTube video player"
+            frameborder="0"
+            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+            allowfullscreen
+          ></iframe>
+        </div>
+        <p>
+          Learn more about the
+          <a href="https://web.uri.edu/rinsfepscor/research/"
+            >NSF-funded RI-C-AIM program</a
+          >.
+        </p>
+        <p>
+          The full dataset used to power this app is available on
+          <a
+            href="https://pricaimcit.services.brown.edu/erddap/search/index.html?page=1&itemsPerPage=1000&searchFor=Buoy+Telemetry"
+            >ERDDAP</a
+          >.
+        </p>
+        <p>
+          Additional information about the data variables and QC tests can be
+          found in the
+          <a
+            href="https://pricaimcit.services.brown.edu/erddap/info/buoy_telemetry_0ffe_2dc0_916e/index.html"
+            >ERDDAP dataset info page</a
+          >.
+        </p>
+      </div>
+    </template>
+  </DashboardCard>
+</template>
+
+<script setup lang="ts">
+import { computed, inject, ref } from "vue";
+import { scaleLinear } from "d3-scale";
+
+import DashboardCard from "@/components/base/DashboardCard.vue";
+import DownloadForm from "@/components/buoy/DownloadForm.vue";
+import ExploreForm from "@/components/buoy/ExploreForm.vue";
+import DateSlider from "@/components/buoy-telemetry/DateSlider.vue";
+import BuoyMap from "@/components/buoy-telemetry/BuoyMap.vue";
+
+import { formatVariable } from "../../../utils/utils";
+import { BuoyStore } from "../../../store/buoy";
+
+const store = inject("store") as BuoyStore;
+
+const endDate = store.maxDateRaw;
+const startDate = new Date(endDate.valueOf() - 5 * 24 * 60 * 60 * 1000);
+
+const variableStr = [
+  "ChlorophyllSurface",
+  "NitrateNSurface",
+  "PhosphateSurface",
+  "AirTemp",
+  "AirPressure",
+  "HumiditySurface",
+  "Precipitation",
+  "WaterTempSurface",
+  "WindDirectionFrom",
+  "WindSpeedAverage",
+];
+const bouyIDs = store.coordinates.map((c) => c.buoyId).join(",");
+
+const samplesRaw = await store.fetchBuoyData({
+  ids: bouyIDs,
+  variables: variableStr.join(","),
+  end: endDate.toISOString(),
+  start: startDate.toISOString(),
+});
+
+const samples = samplesRaw.data.map((s) => {
+  return { ...s, date: new Date(s.time) };
+});
+
+const selectedDate = ref(endDate);
+
+const formattedDate = computed(() => {
+  return new Intl.DateTimeFormat("en-US", {
+    day: "2-digit",
+    month: "2-digit",
+    year: "2-digit",
+    hour: "2-digit",
+    minute: "2-digit",
+  }).format(selectedDate.value);
+});
+
+const selectedSamples = computed(() => {
+  // initialize samples array first
+  const res = [];
+  // for each buoy
+  store.coordinates.forEach((coordinate) => {
+    variableStr.forEach((variable) => {
```

Comment:
I think this entire for loop can be simplified. Right now, it looks like you're doing the following:

1. Finding a sample (`foundSample`)
2. Finding the `nextSample`
3. Reversing the list
4. Getting the `prevSample`
5. Iterpolating between previous and next. 

which is a lot of going through the same array. I think we can go through the array in fewer steps.! Let me know what you think? We can go through this together too! 

Instead, I think we can do something like the following. ⚠️ This code wasn't tested... i just wrote code.
1. Get all of the foundSamples using `filter`

```js
// foundSamples is now a array of all the foundSamples
      const foundSamples = samples.filter((sample) => {
        return (
          sample.date.valueOf() === selectedDate.value.valueOf() &&
          sample.variable === variable &&
          sample.station_name === coordinate.station_name &&
          sample.value !== null
        );
      });
```
2. Interpolate between each sample. 
```js
const newSamples = foundSamples.reduce((res, nextSample, index, array) => {
    if (index !== 0) {
        const prevSample = array[index - 1];

        const interpolator = scaleLinear(
        [prevSample.date.valueOf(), nextSample.date.valueOf()],
        [prevSample.value, nextSample.value]
        );

        const newValue = interpolator(selectedDate.value.valueOf());
        const newSample = {
        ...nextSample,
        date: selectedDate.value,
        value: newValue,
        };

        res.push(newSample);
    }
    return res;

}, []);
```

Reply:
I'll put an issue in so it doesn't delay launching this page any further, and would love to look at it for efficiencies. It does seem like a big block of code that can be refined.

Reply:
I tried to implement it this way and need to debug the `.reduce`